extensions/src/main/java/org/pentaho/platform/plugin/action/ActionInvokerSystemListener.java

```diff
@@ -166,6 +182,7 @@ private void buildPayload( String urlEncodedJson, String enc ) {
         String actionClass = URLDecoder.decode( jsonVars.getString( ActionUtil.INVOKER_ACTIONCLASS ), enc );
         String actionId = URLDecoder.decode( jsonVars.getString( ActionUtil.INVOKER_ACTIONID ), enc );
         actionUser = URLDecoder.decode( jsonVars.getString( ActionUtil.INVOKER_ACTIONUSER ), enc );
+        workItemUid = URLDecoder.decode( jsonVars.getString( ActionUtil.WORK_ITEM_UID ), enc );
```

Comment:
workItemUid needs to be included in the job passed from the main pentaho server and also in the scripts which create the JSON input, otherwise this payload will be looking for key (workItemUid) that doesn't exist.   Now that I'm thinking on it, perhaps we should create some object that allows us to put "extras" in the payload without modifying the scripts.